hrjobcontract/js/test/controllers/modal/modal-contract_test.js

```diff
@@ -0,0 +1,146 @@
+    var ctrl, $rootScope, $controller, $scope, $q, $uibModal, $httpBackend, $uibModalInstanceSpy,
+      $uibModalSpy, entity, ContractDetailsService;
+
+    beforeEach(module('hrjc'));
+
+    beforeEach(inject(function(_$controller_, _$rootScope_, _$httpBackend_, _$q_, _ContractService_,
+      _ContractDetailsService_) {
+      $controller = _$controller_;
+      $rootScope = _$rootScope_;
+      $httpBackend = _$httpBackend_;
+      ContractDetailsService = _ContractDetailsService_;
+      $q = _$q_;
+      entity = {
+        contract: {
+          id: '1',
+          contact_id: "04",
+          deleted: "0",
+          is_current: "1",
+          is_primary: "1"
+        },
+        details: {
+          id: "60",
+          position: "Test-added",
+          title: "Test-added",
+          funding_notes: null,
+          contract_type: "Apprentice",
+          period_start_date: "2017-03-28",
+          period_end_date: null,
+          end_reason: "1",
+          notice_amount: null,
+          notice_unit: null,
+          notice_amount_employee: null,
+          notice_unit_employee: null,
+          location: "Headquarters",
+          jobcontract_revision_id: "60"
+        },
+        hour: {},
+        pay: {},
+        leave: [],
+        health: {},
+        pension: {}
+      };
+
+      // $uibModalInstanceSpy for $uibModalInstance.opened.then()
+      $uibModalInstanceSpy = {
+        opened: {
+          then: function() {
+            jasmine.createSpy('modalInstance.opened');
+          }
+        }
+      };
+
+      // $uibModalSpy for $modal.open()
+      $uibModalSpy = {
+        open: function() {
+          return {
+            result: {
+              then: function(callback) {
+                callback("edit");
+              }
+            }
+          }
+        }
+      };
+
+      $httpBackend.whenGET(/action=validatedates&entity=HRJobDetails/).respond({ success: true });
+      $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond({});
+      $httpBackend.whenPOST(/action=create&entity/).respond({});
+      $httpBackend.whenPOST(/action=replace&entity/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+
+      makeController();
+    }));
+
+
+    describe("Sets contract end date as appropriate", function() {
+
+      beforeEach(function() {
+        createModalSpy();
+        $scope.save();
+        $rootScope.$digest();
+      });
+
+      describe("When period_end_date is null", function() {
+
+        it("Sets contract period_end_date to '' ", function() {
+          expect($scope.entity.details.period_end_date).toBe("");
+        });
+      });
+
+      describe("When period_end_date is not null", function() {
+
+        beforeEach(function() {
+          $scope.entity.details.period_end_date = '02-03-2017';
+        });
+
+        it("Sets contract period_end_date to '02-03-2017' ", function() {
+          expect($scope.entity.details.period_end_date).toBe('02-03-2017');
```

Comment:
What is the purpose of this test? You are setting value in beforeEach and testing with same value, obviously it is going to pass. The actual function is never called.